pkg/api/util/names_test.go

```diff
@@ -111,3 +113,182 @@ func TestComputeName(t *testing.T) {
 		})
 	}
 }
+
+func TestDNSSafeShortenToNCharacters(t *testing.T) {
+	type testcase struct {
+		in        string
+		maxLength int
+		expOut    string
+	}
+
+	tests := []testcase{
```

Comment:
nitpick: could this do with some tests which test different maxLengths? (e.g. maxLength longer than `in`, maxLength set to 0 or 1 or whatever)

Also maybe could do with some different test strings - there aren't any here with numbers, or `-` characters, for example!

Reply:
I added some extra tests